### ISSUE TITLE
Found an issue with using Custom Job Names

### ISSUE THREAD

**Issue by krillgar** [HangfireIO/Hangfire#405]
I'm working on a program that is creating lots of jobs, so I wanted to use custom job names to easily differentiate them.

I am using a combination of a {GUID}|{long}|{string identifier} and it started throwing SQL errors that I traced to the enqueing of the Hangfire job.

I looked through the database, and discovered that the job name is stored in Hangfire.Set.Value. That column has a length of 256, which was fine. The problem comes when the job is added to the Hangfire.Hash table. In there is a column named Key with a max length of 100. Unfortunately, that combines the Set.Value with the Set.Key columns, resulting in a potential length of 350.

**Comment by odinserj** [comment]
>  wanted to use custom job names

Do you mean the [`DisplayNameAttribute`](https://github.com/HangfireIO/Hangfire/issues/144)? However, it does not use `Hangfire.Set` table.

**Comment by krillgar** [comment]
No, I'm manually setting the job ID via the first parameter of RecurringJob.AddOrUpdate<T>.

    RecurringJob.AddOrUpdate<T>({GUID}|{long}|{string}, job => job.Process, {CRON});

The above first parameter is right now 87 characters long. That goes into HangFire.Set.Value, which is nvarchar(256). The Key there is "recurring-jobs".

When that goes into the Hangfire.Hash, the new value is "recurring-job:{GUID}|{long}|{string}", so this new length is 101 attempting to go into a nvarchar(100).

I am not using the `DisplayNameAttribute`.

**Comment by odinserj** [comment]
What are the values of {long} and {string}? You can pass them as arguments to job.Process and use `DisplayNameAttribute` to see human-readable job names, '{N}' placeholders are available:

```csharp
[DisplayName("{0}: {1}")]
public void Process(string name, long id)
```

P.S. Are you sure fire-and-forget and delayed jobs does not satisfy your needs?

**Comment by krillgar** [comment]
The job ID I was trying to use was `4e5c1eae-907b-4f84-8ed9-67e92521b9bc|1|Updated test Scheduled Import Definition Trigger`. The reason for this is so that we can easily identify them not just in the Hangfire UI, but also in the database and in code.

The system that we're creating is going to need to do all the types of jobs that Hangfire allows: fire-and-forget, delayed, and recurring.

Regardless of how I'm creating the jobs, the structure of your database tables is creating a situation where overflow can easily occur while trying to enter data in to a column.

**Comment by zojamax** [comment]
I have same issue with recurring job names. @odinserj are you going to increase [HangFire].[Hash].[Key] length in nearest future? It should be at least 100 + 256 + 1 i.e. [HangFire].[Hash].[Key] = [HangFire].[Set].[Key] + : + [HangFire].[Set].[Value]

**Comment by MrKevHunter** [comment]
We've just fallen off this too, any update?

**Comment by odinserj** [comment]
Please use some identifier only for the recurring job id and use the `DisplayName` attribute as shown above to control the display name of a background or recurring job. Identifiers aren't meant to include strings for display purposes and implemented only to have a unique way of referencing a job. In this case, length of a column should be enough.

Let me know if not, in this case you can try increasing the length of the column, it should work.
